### PR TITLE
Support aarch64-darwin Build

### PR DESCRIPTION
**Pull request by zeme-wana**
This PR introduces support for Apple Silicon (`aarch64-darwin`).

## Building The Haskell Sources

The haskell sources build on `aarch64-darwin` out of the box. 

`aarch64-darwin` has been added to `systems` in `flake.nix` and to `hydraJobs`.

## New Network Environments

Network configurations in marlowe-cardano are defined in [nix/marlowe-cardano/network.nix](https://github.com/input-output-hk/marlowe-cardano/blob/main/nix/marlowe-cardano/networks.nix) 

However, `testnet` comes from the [cardanoLib overlay](https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/default.nix)  in `iohk-nix`, while `preview` and `preprod` come from the [environments output](https://github.com/input-output-hk/cardano-world/blob/d22f50fc77d23e2612ca2b313a098dd0b48834d4/nix/cardano/environments/default.nix) in cardano-world.

But, only the `testnet` configuration is actually used anywhere in the repo, in [compose.nix](https://github.com/input-output-hk/marlowe-cardano/blob/main/nix/marlowe-cardano/compose.nix)

On May 16th cardano-world [removed](https://github.com/input-output-hk/cardano-world/commit/16e25636800c3ae7f4c5aeb8a39baf7a57b1ee7f) `nix/cardano/environments`.

And started wholly relying on the [cardanoLib overlay](https://github.com/input-output-hk/iohk-nix/blob/master/cardano-lib/default.nix) from `iohk-nix`.

So it appears `iohk-nix` is now the SSOT for network environments.

So in this PR we remove the `cardano-world` flake entirely and use `iohk-nix` instead.

However you must check that new configurations are valid. I have included the genesis files in this PR so that you may diff them, and they will be removed from this PR before merging:
```
diff nix/__genesis-files__/byron-genesis-old.json nix/__genesis-files__/byron-genesis-new.json

diff nix/__genesis-files__/shelley-genesis-old.json nix/__genesis-files__/shelley-genesis-new.json
```

## New `cardano-address`

We used to get `cardano-address` from `cardano-world` as well, but now we get it from `cabal.project` instead, so we no longer need `cardano-world` at all. 

`cardano-address --version == 3.12.0` 

Note that using `cardano-address` is optional: we just add it to the shell for convenience. 

Nonetheless, we can now have a native `cardano-address` inside the shell.

## New `cardano-node` and `cardano-cli`

`cardano-node` and `cardano-cli` offer `aarch64-darwin` support starting from `8.2.1` (we are currently on `8.1.2`) but unlike `cardano-address` they are used to run the integration tests.

This PR keeps version 8.1.2 and seamlessly picks `x86_64-darwin` when on `aarch64-darwin`. 

If you are happy to upgrade to `8.2.1` (or greater), we can get `aarch64-darwin` node+cli in this PR

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -190,12 +190,11 @@ package freer-extras
 
 -- marlowe-cli only
 
--- When updated to 3.12.0 or newer, remove the cabal-generator override in nix/pkgs/haskell/haskell.nix
```
Comment (zeme-wana) [inline cabal.project:193]: This comment is obsolete

File: cabal.project [hunk 2]
```diff
@@ -190,12 +190,11 @@ package freer-extras
...
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-addresses
-  tag: b6f2f3cef01a399376064194fd96711a5bdba4a7
```
Comment (zeme-wana) [inline cabal.project:197]: This tag was just non-existent

File: flake.nix [hunk 3]
```diff
@@ -12,12 +12,10 @@
     # Use upstream when https://github.com/nlewo/nix2container/pull/82 is merged
     n2c.url = "github:shlevy/nix2container/no-Size-on-dir";
 
-    marlowe-plutus.url = "github:input-output-hk/marlowe-plutus";
+    marlowe-plutus.url = "github:input-output-hk/marlowe-plutus?ref=aarch64-darwin";
```
Comment (zeme-wana) [inline flake.nix:15]: Will update when https://github.com/input-output-hk/marlowe-plutus/pull/8 is merged

File: nix/__genesis-files__/byron-genesis-new.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (zeme-wana) [inline nix/__genesis-files__/byron-genesis-new.json:1]: See this PR's description for why these files are here

File: nix/marlowe-cardano/cardano-tools.nix [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{ repoRoot, inputs, pkgs, lib, system }:
+
+# TODO Remove this file once cardano-node can build aarch64-darwin.
+
+{
+  cardano-node =
+    if system == "aarch64-darwin"
```
Comment (zeme-wana) [inline nix/marlowe-cardano/cardano-tools.nix:7]: We can get native `aarch64-darwin` cardano-node and cardano-cli if you can upgrade from 8.1.2 to 8.2.1 (or greater)

File: nix/project.nix [hunk 6]
```diff
@@ -5,6 +5,9 @@ let
   cabalProject = pkgs.haskell-nix.cabalProject' ({ config, pkgs, ... }:
     let
       mkIfDarwin = lib.mkIf pkgs.stdenv.hostPlatform.isDarwin;
+      # When `isCross` is `true`, it means that we are cross-compiling the project.
+      # WARNING You must use the `pkgs` coming from cabalProject' for `isCross` to work.
+      isCross = pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform;
```
Comment (zeme-wana) [inline nix/project.nix:10]: Cross compilation on windows is next

File: nix/marlowe-cardano/cardano-tools.nix (hunk 5 above)
Comment (bwbush) [reply nix/marlowe-cardano/cardano-tools.nix:7]: We cannot upgrade beyond 8.1.2 until IOG recommends that stake pool operators do that. The last time I looked, the SPOs were told to continue using 8.1.2.

**Review by bwbush — APPROVED** [review]
The changes look fine, but I don't have a Mac to test this on. Maybe @ramsay-t or @hrajchert could test it on their Macs?

**Pull request merged**